cf-rd/src/main/java/org/eclipse/californium/tools/resources/RDNodeResource.java

```diff
@@ -38,11 +38,15 @@
 	private static final Logger LOGGER = Logger.getLogger(RDNodeResource.class.getCanonicalName());
 	
 	/*
+	 * Use one timer for keeping track of all RDNodeResource's lifeTime expiry tasks.
+	 */
+	private static final Timer lifetimeTimer = new Timer();
```

Comment:
Agreed.  I wanted this to be a minimal change and hence stuck with `Timer` and `TimerTask`.  In the current implementation, each time an end-point would POST on resource /rd, the current timer (and hence its thread) is cancelled and a new one is created.  The improvement is to keep one thread and create/cancel timer tasks as needed.  I should have created the Timer as daemon.

If it ain't a big trouble, in lieu for this pull request I will create another pull-request where `RDNodeResource` is using `ScheduledThreadPoolExecutor`.